File: csharp/high-scores/HighScores.cs
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private readonly List<int> _scores;

    public HighScores(List<int> list) => _scores = list;

    public List<int> Scores() => _scores;

    public int Latest() => _scores.Last();

    public int PersonalBest() => _scores.Max();

    public List<int> PersonalTopThree() => _scores.OrderByDescending(x => x).Take(3).ToList();
}

File: ruby/high-scores/high_scores.rb
class HighScores
  attr_reader :scores, :personal_top_three

  def initialize(scores)
    @scores = scores
    @personal_top_three = scores.max(3)
  end

  def latest
    scores.last
  end

  def personal_best
    personal_top_three.first
  end

  def latest_is_personal_best?
    latest == personal_best
  end
end
